sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -90,6 +90,10 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
 
                 activity.AddTag("http.status_code", message.Response.Status.ToString(CultureInfo.InvariantCulture));
                 activity.AddTag("serviceRequestId", message.Response.Headers.RequestId);
+                if (message.ResponseClassifier.IsErrorResponse(message))
+                {
+                    activity.AddTag("otel.status_code", "ERROR");
```

Comment:
This allows us to remove the status calculation logic from the AppInsights. PR to follow.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/19982